libs/api/domains/license-service/src/lib/client/adr-license-client/adrLicenseMapper.ts

```diff
@@ -0,0 +1,108 @@
+import { AdrDto } from '@island.is/clients/adr-and-machine-license'
+import {
+  GenericLicenseDataFieldType,
+  GenericUserLicensePayload,
+} from '../../licenceService.type'
+import { FlattenedAdrDto } from './genericAdrLicense.type'
+
+export const parseAdrLicensePayload = (
+  license: AdrDto,
+): GenericUserLicensePayload | null => {
+  if (!license) return null
+
+  const parseAdrLicenseResponse = () => {
+    const { adrRettindi, ...rest } = license
+
+    const flattenedAdr: FlattenedAdrDto = { ...rest, adrRettindi: [] }
+
+    adrRettindi?.forEach((field) => {
+      const heiti =
+        field.heiti && field.heiti.length
+          ? field.heiti
+          : field.flokkur?.split(',').map((f) => ({
+              flokkur: f.trim(),
+              heiti: '',
+            }))
+      heiti?.forEach((item) => {
+        flattenedAdr.adrRettindi?.push({
+          flokkur: item.flokkur,
+          grunn: field.grunn,
+          tankar: field.tankar,
+          heiti: item.heiti,
+        })
+      })
+    })
+    return flattenedAdr
+  }
+
+  const parsedResponse = parseAdrLicenseResponse()
+
+  const data = [
```

Comment:
sync labels w design
https://www.figma.com/file/kzvPtwIM25MrBmoKOhEkji/%C3%8Dsland.is-%E2%80%93-App?node-id=2747%3A27519